services/messageService.js

```diff
@@ -0,0 +1,51 @@
+import { AsyncStorage } from 'react-native';
+
+class MessageService {
+  constructor(newMessagesCallback) {
+    this.messages = {};
+    this.newMessagesCallback = newMessagesCallback;
+  }
+
+  static async cacheMessages(contact, messages) {
```

Comment:
Should we move this into a util class instead of a static member function?

Reply:
I think in general we should, but since this refactor is a mid-point on the way to redux (which will ultimately handle the caching logic), it's fine to leave as a static function.